Doppler.HtmlEditorApi/Storage.DapperProvider/Repository.cs

```diff
@@ -102,6 +102,14 @@ public async Task SaveCampaignContent(string accountName, ContentData contentRow
             _ => throw new NotImplementedException($"Unsupported campaign content type {contentRow.GetType()}")
         };
 
-        await query.ExecuteAsync(queryParams);
+        await upsertContentQuery.ExecuteAsync(queryParams);
+
+        var updateCampaignStatusQuery = new UpdateCampaignStatusDbQuery(_dbContext);
+
+        await updateCampaignStatusQuery.ExecuteAsync(new UpdateCampaignStatusDbQuery.Parameters(
+            setCurrentStep: 2,
+            setHtmlSourceType: 2,
+            whenIdCampaignIs: contentRow.campaignId,
+            whenCurrentStepIs: 1));
```

Comment:
@MarianoHerrera what do you think about this approach?

I think that it is not totally optimized (we are doing 3 queries to the database) for a PUT method, but compared with Doppler implementation it is really cheaper.

On the other hand, we have no transactions, in my opinion, if something goes wrong, the database state will be acceptable, but this is arguable. What do you think?

Reply:
We talked with @MarianoHerrera, we agree that using transactions would be desirable. I would work on that, but with less priority.

Reply:
**UPDATE:**

@MarianoHerrera I tested it with Doppler, and if the third step is not executed, Doppler works fine, I mean, works in the same way that when the content does not exist.

So, using transactions is not necessary here, but anyway I will implement them for further scenarios.

Reply:
Marking this conversation as solved.